File: MPChartExample/src/com/xxmassdeveloper/mpchartexample/custom/MyValueFormatter.java
package com.xxmassdeveloper.mpchartexample.custom;

import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.formatter.FormattedStringCache;
import com.github.mikephil.charting.formatter.ValueFormatter;
import com.github.mikephil.charting.utils.ViewPortHandler;

import java.text.DecimalFormat;

public class MyValueFormatter implements ValueFormatter {

    private FormattedStringCache<Integer, Float> mFormattedStringCache;
    
    public MyValueFormatter() {
        mFormattedStringCache = new FormattedStringCache<>(new DecimalFormat("###,###,###,##0.0"));
    }

    @Override
    public String getFormattedValue(float value, Entry entry, int dataSetIndex, ViewPortHandler viewPortHandler) {
        return mFormattedStringCache.getFormattedString(value, dataSetIndex) + " $";
    }
}

File: MPChartExample/src/main/java/com/xxmassdeveloper/mpchartexample/custom/MyValueFormatter.kt
package com.xxmassdeveloper.mpchartexample.custom

import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.formatter.IValueFormatter
import com.github.mikephil.charting.utils.ViewPortHandler
import java.text.DecimalFormat

class MyValueFormatter : IValueFormatter {
    private val decimalFormat = DecimalFormat("###,###,###,##0.0")

    override fun getFormattedValue(value: Float, entry: Entry?, dataSetIndex: Int, viewPortHandler: ViewPortHandler?): String? {
        return decimalFormat.format(value.toDouble()) + " $"
    }
}
